nixos/modules/services/mail/maddy.nix

```
@@ -318,6 +390,17 @@ in {
         $(primary_domain) = ${cfg.primaryDomain}
         $(local_domains) = ${toString cfg.localDomains}
         hostname ${cfg.hostname}
+
+        ${if (cfg.tls.loader == "file") then ''
+          tls file ${concatStringsSep " " (
+            map (x: x.certPath + " " + x.keyPath
+          ) cfg.tls.certificates)} ${optionalString (cfg.tls.extraConfig != null) ''
```

Or just don't check if it's set and always include the empty parentheses?
```suggestion
          ) cfg.tls.certificates)} ${optionalString (cfg.tls.extraConfig != "") ''
```